docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1280 @@
+the parameter name is specified in the **required** `name` attribute.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// <para>The main <c>Math</c> class.</para>
+/// <para>Contains all methods for performing basic math functions.</para>
+/// </summary>
+public class Math
+{
+    /// <summary>
+    /// Adds two doubles <paramref name="a"/> and <paramref name="b"/> and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two doubles.
+    /// </returns>
+    /// <exception cref="System.OverflowException">Thrown when one parameter is max and the other is greater than zero.</exception>
+    /// See <see cref="Math.Add(int, int)"/> to add integers.
+    /// <param name="a">A double precision number.</param>
+    /// <param name="b">A double precision number.</param>
+    public static double Add(double a, double b)
+    {
+        if ((a == double.MaxValue && b > 0) || (b == double.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+}
+```
+
+### &lt;typeparamref&gt;
+
+You use `<typeparamref>` tag just like the `<paramref>` tag but for generic type or method declarations to describe a generic parameter.
+You can use the same generic method you previously created.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// <para>The main <c>Math</c> class.</para>
+/// <para>Contains all methods for performing basic math functions.</para>
+/// </summary>
+public class Math
+{
+    /// <summary>
+    /// Checks if an IComparable <typeparamref name="T"/> is greater than another.
+    /// </summary>
+    /// <typeparam name="T">A type that inherits from the IComparable interface.</typeparam>
+    public static bool GreaterThan<T>(T a, T b) where T : IComparable
+    {
+        return a.CompareTo(b) > 0;
+    }
+}
+```
+
+### &lt;list&gt;
+
+You use the `<list>` tag to format documentation information as an ordered list, unordered list or table.
+You'll make an unordered list of every math operation your `Math` library supports, you also get stub samples for ordered list and table.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// <para>The main <c>Math</c> class.</para>
+/// <para>Contains all methods for performing basic math functions.</para>
+/// <list type="bullet">
+/// <item>
+/// <term>Add</term>
+/// <description>Addition Operation</description>
+/// </item>
+/// <item>
+/// <term>Subtract</term>
+/// <description>Subtraction Operation</description>
+/// </item>
+/// <item>
+/// <term>Multiply</term>
+/// <description>Multiplication Operation</description>
+/// </item>
+/// <item>
+/// <term>Divide</term>
+/// <description>Division Operation</description>
+/// </item>
+/// </list>
+/// </summary>
+
+/// <summary>
```

Comment:
I'd delete this list. It's not part of the library, and you can say that by not specifying the bullet type, you create an ordered list.